pymagine/colour_filters.py

```diff
@@ -25,3 +28,110 @@ def colour_filters(image, tone="sepia"):
+  elif tone.lower() == 'negative':
+      image_2d = image[:, :, 0]
+      
+      negative_image = image_2d.copy()
+      negative_image = 255 - image_2d
+      final_image = plt.imshow(negative_image, cmap="gray")
+      
+  elif tone.lower() == 'red_tone':
+      transformed_red = red_pixel*0.75
+      transformed_blue = blue_pixel*0.125
+      transformed_green = green_pixel*0.125
+
+      transformed_red = (np.where(transformed_red > 255, 255, transformed_red)).astype(int)
+      transformed_green = (np.where(transformed_green > 255, 255, transformed_green)).astype(int)
+      transformed_blue = (np.where(transformed_blue > 255, 255, transformed_blue)).astype(int)
+      
+      final_image_array = np.dstack((np.reshape(transformed_red, n), np.reshape(transformed_green, n), np.reshape(transformed_blue, n)))
+      final_image = plt.imshow(final_image_array)
+      
+  elif tone.lower() == 'blue_tone':
+      transformed_red = red_pixel*0.12
+      transformed_blue = blue_pixel*0.70
+      transformed_green = green_pixel*0.17
+
+      transformed_red = (np.where(transformed_red > 255, 255, transformed_red)).astype(int)
+      transformed_green = (np.where(transformed_green > 255, 255, transformed_green)).astype(int)
+      transformed_blue = (np.where(transformed_blue > 255, 255, transformed_blue)).astype(int)
+      
+      final_image_array = np.dstack((np.reshape(transformed_red, n), np.reshape(transformed_green, n), np.reshape(transformed_blue, n)))
+      final_image = plt.imshow(final_image_array)
+      
+  elif tone.lower() == 'green_tone':
+      transformed_red = red_pixel*0.12
+      transformed_blue = blue_pixel*0.17
+      transformed_green = green_pixel*0.70
+
+      transformed_red = (np.where(transformed_red > 255, 255, transformed_red)).astype(int)
+      transformed_green = (np.where(transformed_green > 255, 255, transformed_green)).astype(int)
+      transformed_blue = (np.where(transformed_blue > 255, 255, transformed_blue)).astype(int)
+      
+      final_image_array = np.dstack((np.reshape(transformed_red, n), np.reshape(transformed_green, n), np.reshape(transformed_blue, n)))
+      final_image = plt.imshow(final_image_array)
+  
+  else:
+      transformed_red = red_pixel*0.393 + green_pixel*0.769 + blue_pixel*0.189
+      transformed_blue = red_pixel*0.272 + green_pixel*0.534 + blue_pixel*0.131
+      transformed_green = red_pixel*0.349 + green_pixel*0.686 + blue_pixel*0.168
+
+      transformed_red = (np.where(transformed_red > 255, 255, transformed_red)).astype(int)
+      transformed_green = (np.where(transformed_green > 255, 255, transformed_green)).astype(int)
+      transformed_blue = (np.where(transformed_blue > 255, 255, transformed_blue)).astype(int)
+      
+      n = image[:,:,0].shape
+      
+      final_image_array = np.dstack((np.reshape(transformed_red, n), np.reshape(transformed_green, n), np.reshape(transformed_blue, n)))
+      final_image = plt.imshow(final_image_array)
+  
+  return final_image
```

Comment:
```suggestion
  cv2.imwrite("colour_filters.jpeg", final_image)
```

Reply:
cv2.imwrite will not work with the plt.imshow object saved in `final_image`. I will have to edit some more portions of the file to save the image